app/routes.py

```diff
@@ -1,2 +1,239 @@
+
+        goal.title = form_data["title"]
+        
+        db.session.commit()
+
+        return jsonify({
+            "goal":{
+                "id":goal.goal_id,
+                "title":goal.title,
+            }
+        }),200
+
+    elif request.method == "DELETE":
+        db.session.delete(goal)
+        db.session.commit()
+
+        return jsonify({
+            "details": f'Goal {goal.goal_id} "{goal.title}" successfully deleted'
+        }),200
+        
+
+@goals_bp.route("/<goal_id>/tasks", methods=["POST","GET"])
+def post_tasked_goal(goal_id):
+
+        goal = Goal.query.get(goal_id)
+
+        if goal == None:
+            return (""), 404
+
+        if request.method == "POST":
+            request_body = request.get_json()
+          
+            tasks_instances= []
+            for task_id in request_body["task_ids"]:
+                tasks_instances.append(Task.query.get(task_id))
+            
+            goal.tasks = tasks_instances
+            
+            db.session.commit()
+
+            task_ids = []
+            for task in goal.tasks:
+                task_ids.append(task.task_id)
+             
+            response_body = {
+                        "id": goal.goal_id,
+                        "task_ids": task_ids
+                    }
+                    
+            return jsonify(response_body), 200
+
+        if request.method == "GET":
+            tasks_response =[]
+            for task in goal.tasks:
+                tasks_response.append({
+                    "id": task.task_id,
+                    "goal_id": task.goal_id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": bool(task.completed_at)
+                })
+            response_body = {
+                "id": goal.goal_id,
+                "title": goal.title,
+                "tasks" : tasks_response
+            }
+            return jsonify(response_body), 200
+
+
+@tasks_bp.route("",methods=["GET","POST"])
+def handle_task():
+    if request.method == "GET": 
+
+        sorting_task = request.args.get('sort') 
+        list = None
+        if sorting_task == "desc":
+            list = Task.query.order_by(Task.title.desc()) # descending method
+        elif sorting_task == "asc":
+            list = Task.query.order_by(Task.title.asc()) # ascending method
+        else: 
+            list = Task.query.all()
+        tasks_response = []
+        for task in list:
+            tasks_response.append({
+            "id": task.task_id,
+            "title": task.title,
+            "description": task.description,
+            "is_complete" : False
+            if task.completed_at == None else True
+            })
+
+        return jsonify(tasks_response)
+
+    elif request.method == "POST":
+        request_body = request.get_json()
+        if "title" not in request_body or "description" not in request_body or "completed_at" not in request_body:
```

Comment:
👍 

Good job enforcing the POST data requirements. We should be doing similar checks when PUTting a task as well. So we could also think about moving checks like this into validation helpers so that they are easier to reuse elsewhere.

We could even think about adding a class method to `Task` to create a new `Task` using the dictionary from a request body

```python
    @classmethod
    def from_dict(values):
        # create a new task, and set the model values from the values passed in
        # be sure to validate that all required values are present, we could return `None` or raise an error if needed
        return new_task
```